Libraries/LibHTTP/Job.cpp

```diff
@@ -151,7 +151,50 @@ void Job::on_socket_connected()
         ASSERT(can_read());
 
         read_while_data_available([&] {
-            auto payload = receive(64 * KB);
+            auto read_size = 64 * KB;
+            if (m_current_chunk_remaining_size.has_value()) {
+            read_chunk_size:;
+                auto remaining = m_current_chunk_remaining_size.value();
+                if (remaining == -1) {
+                    // read size
+                    auto size_data = read_line(PAGE_SIZE);
+                    auto size_lines = StringView { size_data.data(), size_data.size() }.lines();
+                    if (size_lines.size() == 0) {
+                        dbg() << "Job: Reached end of stream";
+                        read_line(PAGE_SIZE);
+                    } else {
+                        String size_string = size_lines[0];
+                        auto size = strtoul(size_string.characters(), nullptr, 16);
```

Comment:
What if it's not a valid number?

Reply:
hmmm...I suppose we should fail then.

now how does that endptr work...?

Reply:
> how does that endptr work

It actually doesn't.
I'll add the fix to this PR, if that's alright?

Reply:
Ofc!